### PR TITLE
DALA 4684: Add 7 simple fields to SfDR data set - Deka RfP

### PR DESCRIPTION
**Pull request by LaurinStrelow**
# Pull Request DALA 4684
`Add 7 simple fields to SfDR data set - Deka RfP`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
  - [x] If this PR includes work on the frontend, at least one `@ts-nocheck` is removed. Additionally, there should not be any `@ts-nocheck` in files modified by this PR. If no `@ts-nocheck` are left: Celebrate :tada: :confetti_ball: type-safety and remove this entry. 
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [x] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [x] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [x] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on the clone server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
  - [x] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [ ] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [ ] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [ ] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on dev server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team

### CODE REVIEW

File: dataland-internal-storage/src/main/kotlin/db/migration/V22__IntroduceNewFieldsInSfdr.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+     * Migrates sfdr data to the new sfdr data model.
+     */
+    fun migrateBoardFields(dataTableEntity: DataTableEntity) {
+        val dataset = dataTableEntity.dataJsonObject
+
+        val socialObject = dataset.getJSONObject("social") ?: return
+
+        val socialAndEmployeeMattersObject = socialObject.getJSONObject("socialAndEmployeeMatters") ?: return
+
+        val femaleBoardMembers = socialAndEmployeeMattersObject.remove("femaleBoardMembers")
```
Comment (SiegfriedSobkowiak) [inline dataland-internal-storage/src/main/kotlin/db/migration/V22__IntroduceNewFieldsInSfdr.kt:31]: Man könnte hier auch eine Map anlegen mit oldName : newName und dann einfach über alle Keys gehen und die Ersetzung machen.

File: dataland-internal-storage/src/main/kotlin/db/migration/V22__IntroduceNewFieldsInSfdr.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        socialAndEmployeeMattersObject.put("femaleBoardMembersSupervisoryBoard", femaleBoardMembers)
+
+        val maleBoardMembers = socialAndEmployeeMattersObject.remove("maleBoardMembers")
```
Comment (SiegfriedSobkowiak) [inline dataland-internal-storage/src/main/kotlin/db/migration/V22__IntroduceNewFieldsInSfdr.kt:34]: Man braucht auf jeden Fall noch einen check ob der Key den man ersetzen möchte in dem json tatsächlich vorkommt. In den Daten auf Produktion kann es sein das Felder die erst später dazugekommen sind nicht in den DB strings enthalten sind.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/frameworks/sfdr/model/environmental/greenhouseGasEmissions/SfdrEnvironmentalGreenhouseGasEmissions.kt [hunk 3]
```diff
@@ -56,6 +62,18 @@ data class SfdrEnvironmentalGreenhouseGasEmissions(
     @field:Valid()
     val ghgIntensityInTonnesPerMillionEURRevenue: ExtendedDataPoint<BigDecimal?>? = null,
 
+    @field:Valid()
+    val ghgIntensityScope1InTonnesPerMillionEURRevenue: ExtendedDataPoint<BigDecimal?>? = null,
```
[the remark this answers is not in the document]
Comment (LaurinStrelow) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/frameworks/sfdr/model/environmental/greenhouseGasEmissions/SfdrEnvironmentalGreenhouseGasEmissions.kt:66]: Das sollte konsistent sein und trifft auf die anderen Felder auch zu. Wir wollen eigentlich auch keine Mio. Euro als Einheit haben. Deswegen wäre es vmtl. am besten den Tooltip anzupassen. Wie siehst du das?

File: dataland-internal-storage/src/main/kotlin/db/migration/V22__IntroduceNewFieldsInSfdr.kt (hunk 1 above)
Comment (LaurinStrelow) [reply dataland-internal-storage/src/main/kotlin/db/migration/V22__IntroduceNewFieldsInSfdr.kt]: Das ist im neusten commit umgesetzt.

File: dataland-internal-storage/src/main/kotlin/db/migration/V22__IntroduceNewFieldsInSfdr.kt (hunk 2 above)
Comment (LaurinStrelow) [reply dataland-internal-storage/src/main/kotlin/db/migration/V22__IntroduceNewFieldsInSfdr.kt]: Im Prinzip sollte das ok sein. Remove gibt null zurück, wenn der key nicht exisitiert. Ich habe im neusten commit die Logik leicht verändert, dass in diesem Fall der key nicht weiter beachtet wird.

**Pull request merged**